runtime/src/main/scala/akka/grpc/internal/ClientState.scala

```diff
@@ -94,25 +98,41 @@ final class ClientState(settings: GrpcClientSettings, channelFactory: GrpcClient
     }
   }
 
-  private def create(): InternalChannel = {
-    val internalChannel: InternalChannel = channelFactory(settings)
-    internalChannel.done.onComplete {
+  private def create(): Future[InternalChannel] =
+    Patterns.retry(
+      () => channelFactory(settings),
```

Comment:
So we retry creation for any type of exception but once it has been re-created we only retry for the two specific ones

Reply:
I agree that's a bit weird. my reasoning was that since we delay between attempts it's not so heavy - for the re-creation we don't have a delay yet so non-transient errors would 'spin'. I think once we add a delay there it might make sense to retry for all exceptions there, too.